python/ray/autoscaler/node_launcher.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import threading
+
+from ray.autoscaler.tags import (TAG_RAY_LAUNCH_CONFIG, TAG_RAY_NODE_STATUS,
+                                 TAG_RAY_NODE_TYPE, TAG_RAY_NODE_NAME,
+                                 STATUS_UNINITIALIZED, NODE_TYPE_WORKER)
+from ray.autoscaler.util import hash_launch_conf
+
+logger = logging.getLogger(__name__)
+
+
+class NodeLauncher(threading.Thread):
```

Comment:
Might be nice to convert this to asyncio at some point instead of using threads now that we're python 3+

Reply:
You actually need real threads here since they do blocking IO with third party libraries.